### PR TITLE
Drain/resume toggle (v0.4.0)

### PR DESCRIPTION
**Pull request by brtknr**

### CODE REVIEW

**Comment by brtknr** [comment]
Goes hand in hand with this PR which invokes drain/resume based on the different between current state obtained from stack output and the desired state from the config file: https://github.com/stackhpc/p3-appliances/pull/49

File: tasks/drain.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# Ansible tasks to drain a Slurm compute node. Waits for the compute node to be
+# drained for up to a day by default.
+#
+# Variables:
+# - node_to_drain: compute node to drain
+# - drain_timeout: seconds to wait for node to drain, default is 86400.
+
+- name: Drain compute node
+  become: yes
+  command: "scontrol update nodename={{ inventory_hostname }} state=DRAIN reason='maintenance'"
```
Comment (oneswig) [inline tasks/drain.yml]: Just wondering if we should delegate things like this to `openhpc_slurm_control_host`

Comment (priteau) [reply tasks/drain.yml]: Isn't this what's being done in `tasks/main.yml`?

File: README.md [hunk 2]
```diff
@@ -23,14 +23,35 @@ Role Variables
 * `control`: whether to enable control host
 * `batch`: whether to enable compute nodes 
 * `runtime`: whether to enable OpenHPC runtime
+* `drain`: whether to drain a compute nodes
```
Comment (priteau) [inline README.md]: `a compute node` or `compute nodes`

File: README.md [hunk 3]
```diff
@@ -23,14 +23,35 @@ Role Variables
...
+* `resume`: whether to resume a compute nodes
```
Comment (priteau) [inline README.md]: `a compute node` or `compute nodes`

File: tasks/drain.yml (hunk 1 above)
Comment (brtknr) [reply tasks/drain.yml]: I added the delegation after Stig pointed this out :)

File: README.md (hunk 2 above)
Comment (brtknr) [reply README.md]: Thanks Pierre!

File: README.md [hunk 4]
```diff
@@ -23,14 +23,35 @@ Role Variables
...
+And an Ansible inventory as this:
+
+    [openhpc_login]
+    openhpc-login-0 ansible_host=10.60.253.40 ansible_user=centos
+
+    [openhpc_compute]
+    openhpc-compute-0 ansible_host=10.60.253.31 ansible_user=centos
+    openhpc-compute-1 ansible_host=10.60.253.32 ansible_user=centos
+
+    [cluster_control:children]
```
Comment (priteau) [inline README.md]: Do we need to create the `cluster_login` group in the inventory too?

File: tasks/drain.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+- name: Drain compute node
+  command: "scontrol update nodename={{ inventory_hostname }} state=DRAIN reason='maintenance'"
+  when: inventory_hostname not in drained_nodes_results.stdout_lines
+
+- name: Check node has drained
+  command: "sinfo --noheader --Node --format='%N' --states=DRAINED"
+  register: drained_nodes
+  until: "inventory_hostname in drained_nodes.stdout_lines"
+  delay: 10
```
Comment (priteau) [inline tasks/drain.yml]: We can turn `10` into a variable with a default value, to reuse it everywhere in the tasks.

File: README.md (hunk 4 above)
Comment (oneswig) [reply README.md]: That may be good, for consistency.

File: README.md [hunk 6]
```diff
@@ -53,19 +74,44 @@ To deploy, create a playbook which looks like this:
...
-    [openhpc_compute]
-    openhpc-compute-0 ansible_host=10.60.253.33 ansible_user=centos
-
-    [cluster_control:children]
-    openhpc_login
+    ---
+    - hosts: openstack
+      gather_facts: false
+      
+      roles:
```
Comment (oneswig) [inline README.md]: Some commenting on the process used here would be good.

File: tasks/main.yml [hunk 7]
```diff
@@ -1,10 +1,18 @@
...
-  when: openhpc_enable.batch | bool
+  when: openhpc_enable.batch | default(false) | bool
 
 - include: runtime.yml
-  when: openhpc_enable.runtime | bool
+  when: openhpc_enable.runtime | default(false) | bool
+
+- include: drain.yml
+  when: openhpc_enable.drain | default(false) | bool
+  delegate_to: "{{ openhpc_slurm_control_host }}"
```
Comment (oneswig) [inline tasks/main.yml]: Loving your work here BK

File: tasks/drain.yml (hunk 5 above)
Comment (brtknr) [reply tasks/drain.yml]: Done

File: README.md (hunk 6 above)
Comment (brtknr) [reply README.md]: Done

File: tasks/main.yml (hunk 7 above)
Comment (brtknr) [reply tasks/main.yml]: THanks :)

**Pull request merged**